adapters/adyoulike/adyoulike.go

```diff
@@ -0,0 +1,159 @@
+package adyoulike
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+
+	"github.com/buger/jsonparser"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+)
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
+	return &AdYouLikeAdapter{
+		endpoint: config.Endpoint,
+	}, nil
+}
+
+type AdYouLikeAdapter struct {
+	endpoint string
+}
+
+func (adapter *AdYouLikeAdapter) MakeRequests(
+	openRTBRequest *openrtb.BidRequest,
+	reqInfo *adapters.ExtraRequestInfo,
+) (
+	requestsToBidder []*adapters.RequestData,
+	errs []error,
+) {
+	var err error
+	var tagID string
+
+	if len(openRTBRequest.Imp) > 0 {
+		var imp = &openRTBRequest.Imp[0]
+		var bidderExt adapters.ExtImpBidder
+		if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			errs = append(errs, newBadInputError(err.Error()))
+		}
+
+		var ttxExt openrtb_ext.ExtImpAdyoulike
+		if err := json.Unmarshal(bidderExt.Bidder, &ttxExt); err != nil {
+			errs = append(errs, newBadInputError(err.Error()))
+		}
+	}
+
+	if len(openRTBRequest.Imp) == 0 {
+		errs = append(errs, newBadInputError("No impression in the bid request"))
+	}
+
+	reqCopy := *openRTBRequest
+	reqCopy.Imp = []openrtb.Imp{}
+	for ind, imp := range openRTBRequest.Imp {
+		reqCopy.Imp = append(reqCopy.Imp, imp)
+
+		tagID, err = jsonparser.GetString(reqCopy.Imp[ind].Ext, "bidder", "placement")
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		reqCopy.Imp[ind].TagID = tagID
```

Comment:
This is fine, but please be aware I think it is modifying the original request which you seem to be avoiding with making the request copy. You may want to make a imp copy too.

Reply:
Thanks will do.

Reply:
Will check that, thanks

Reply:
Checked :  the append:   append(reqCopy.Imp, imp)   is filling the array by value, making a copy of Imp. No change on original object though.